specification/columns/commitmentdiscountquantity.md

```diff
@@ -0,0 +1,64 @@
+# Commitment Discount Quantity
+
+Commitment Discount Quantity is the amount of purchased, used, or unused [*Commitment Discount Units*](#glossary:commitmentdiscountunit) applied to a [*commitment discount's*](#commitment-discount) [*row*](#glossary:row). For a purchase, the Commitment Discount Quantity is either the total or apportioned number of *Commitment Discount Units* over a *commitment discount's* [*term*](#glossary:term). For committed usage, the Commitment Discount Quantity is either the number of *Commitment Discount Units* consumed by a [*row*](glossary:#row) that is covered by a *commitment discount* or is the unused portion of a *commitment discount* over a [*charge period*](#glossary:chargeperiod). The CommitmentDiscountQuantity column only applies to *commitment discounts* and not [*negotiated discounts*](#glossary:negotiated-discount).
+
+For purchases, the Commitment Discount Quantity helps customers track how many total *Commitment Discount Units* are available across *commitment discounts* at various points in time, and for committed usage, it helps customers track how efficiently their *commitment discounts* apply to their workloads. The *commitment discount's* utilization rate is used to measure this efficiency and can be derived by dividing the used portion of CommitmentDiscountQuantity by the sum of CommitmentDiscountQuantity for the *charge period*. As an example, a SQL statement calculating a commitment discount's utilization rate might be:
+
+```sql
+SELECT (
+    SELECT SUM(CommitmentDiscountQuantity) FROM <dataset> WHERE ChargeCategory = 'Usage'
+    AND CommitmentDiscountStatus = 'Used' AND CommitmentDiscountId = '<commitment-discount-id>'
+    /
+    SELECT SUM(CommitmentDiscountQuantity) FROM <dataset> WHERE ChargeCategory = 'Usage'
+    AND CommitmentDiscountId = '<commitment-discount-id>'
+)
+```
+
+With this information, customers can analyze their existing *commitment discount* strategy to make more informed decisions on additional purchases.
+
+**Important:** When aggregating Commitment Discount Quantity for utilization calculations, it's important to exclude either one-time or recurring *commitment discount* purchases (i.e. when *ChargeCategory* is *Purchase*) that are paid to cover future eligible charges (e.g., *Commitment Discount*). This exclusion helps prevent double counting of these charges in the aggregation.
```

Comment:
In this paragraph, we are highlighting the partial-upfront payment scenario because that's when we can expect both one-time and recurring charges, correct? If that's the case, I suggest we specify that this applies **specifically to partial-upfront payments**.

Additionally, based on our discussions and the sample data, I would expect that the following applies: 

> the sum of quantities in purchase-related records should match the sum of quantities in usage-related records for the same CommitmentDiscountId over the commitment term. 

However, **I don't believe there is any double counting in the purchase-related records.**

Once again, a quote from the Commitment Discount Appendix:

> Half of the commitment is billed once during the first charge period of the term for $4,380 (derived as 24 hours * 182.5 days * $1.00), and the other half is billed across each charge period over the term, derived as ($1.00 * 8,760 hours * 0.5).

It’s possible that I may have misinterpreted something, or we may need to adjust the sample data, i.e. appendix.